llarp/service/endpoint_util.hpp

```diff
@@ -41,6 +41,23 @@ namespace llarp
       GetConvoTagsForService(const ConvoMap& sessions, const Address& addr,
                              std::set< ConvoTag >& tags);
     };
+
+    template < typename Endpoint_t >
+    static absl::optional< path::Path::UniqueEndpointSet_t >
+    GetManyPathsWithUniqueEndpoints(Endpoint_t* ep, size_t N, size_t tries = 10)
+    {
```

Comment:
Would it not suffice to return an empty set rather than an empty optional? More specifically, what's the difference between (1) a non-empty optional containing an empty set and (2) an empty optional?

Reply:
this is a good point

Reply:
changed